test/java/seedu/addressbook/data/AddressBookTest.java

```diff
@@ -0,0 +1,359 @@
+package seedu.addressbook.data;
+
+import static org.junit.Assert.*;
+
+import java.util.Iterator;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.person.UniquePersonList.DuplicatePersonException;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.data.tag.UniqueTagList.DuplicateTagException;
+import seedu.addressbook.data.tag.UniqueTagList.TagNotFoundException;
+
+/**
+ * we check, in particular, for the two guarantees specified by the AddressBook class
+ *  - Every tag found in every person will also be found in the tag list.
+ *  - The tags in each person point to tag objects in the master list. (== equality)
```

Comment:
By convention this class tests the AddressBook class. So there is no need to explain what functionality it tests. It should test _all_ functionality of the AddressBook class. Tests can be organized based on public methods of the class such that for each public method there is one or more test methods.